Source/WebCore/platform/graphics/cg/ImageBufferIOSurfaceBackend.cpp

```diff
@@ -162,11 +162,14 @@ RefPtr<NativeImage> ImageBufferIOSurfaceBackend::copyNativeImageForDrawing(Graph
         // The destination backend needs to read the actual pixels. Returning non-refence will
         // copy the pixels and but still cache the image to the context. This means we must
         // return the reference or cleanup later if we return the non-reference.
-        return NativeImage::create(adoptCF(CGIOSurfaceContextCreateImageReference(m_surface->ensurePlatformContext())));
+        if (auto image = adoptCF(CGIOSurfaceContextCreateImageReference(m_surface->ensurePlatformContext()))) {
+            CGImageSetCachingFlags(image.get(), kCGImageCachingTransient);
```

Comment:
I think this needs a comment saying that we're trying to avoid a cache maintained by CG (otherwise, a reader will just see the `#if CACHE_SUBIMAGES` call site and assume we're only trying to avoid our own cache, but using a CG flag to do it.